datadog/resource_datadog_dashboard.go

```diff
@@ -4903,8 +4903,10 @@ func buildDatadogEventQuery(data map[string]interface{}) datadogV1.FormulaAndFun
 	eventQuery.SetIndexes(indexes)
 
 	if terraformSearches, ok := data["search"].([]interface{}); ok && len(terraformSearches) > 0 {
-		terraformSearch := terraformSearches[0].(map[string]interface{})
-		eventQuery.Search = datadogV1.NewFormulaAndFunctionEventQueryDefinitionSearch(terraformSearch["query"].(string))
+		terraformSearch, ok := terraformSearches[0].(map[string]interface{})
+		if ok {
+			eventQuery.Search = datadogV1.NewFormulaAndFunctionEventQueryDefinitionSearch(terraformSearch["query"].(string))
+		}
```

Comment:
I'm not sure why having this config:

```
query {
    event_query {
        data_source = "rum"
        indexes     = ["*"]
        name        = "query1"
        compute {
            aggregation = "count"
        }
        search {
            query = ""
        }
    }
}
```

Did not cause a crash before (where search.query is an empty string), but I caught it this time around and fixed it with these changes. However, this fix seems to make terraform think there was a change on `search` if I repeatedly run `terraform apply`. What is the best way to prevent that? My fix here is referenced from [Handle crash in timeseries_definition #863](https://github.com/DataDog/terraform-provider-datadog/pull/863/files#diff-ad347e3f947d02aa6327b5169164d42bcf32d07d467a6af8d7dcf77ced0cf62d)

Reply:
Conversely, is there a clean way of having Terraform send `query = ""`?

Reply:
Worked around this by first flattening `search.query` to `search_query`; but the [latest changes](https://github.com/DataDog/terraform-provider-datadog/pull/951/commits/a3b25c38b699b11c7c2b3b90b0ba1f28b8491c74) implements a validator which catches an empty string value for `search.query`.